### PR TITLE
Gh 1297 federatedstore gaffer cache save graphs

### PR DESCRIPTION
**Pull request by m55624**

### CODE REVIEW

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java [hunk 1]
```diff
@@ -47,165 +49,175 @@
...
-        FederatedStore store = new FederatedStore();
-
         Schema expectedSchema = new Schema.Builder().build();
-        String expectedGraphId = "testGraphID";
 
-        StoreProperties storeProperties = new StoreProperties();
-        storeProperties.set("gaffer.store.class", "uk.gov.gchq.gaffer.federatedstore.FederatedStore");
+        StoreProperties graphStoreProperties = new StoreProperties();
+        graphStoreProperties.set(StoreProperties.STORE_CLASS, FEDERATEDSTORE_CLASS_STRING);
+        graphStoreProperties.set(CacheProperties.CACHE_SERVICE_CLASS, CACHE_SERVICE_CLASS_STRING);
```
Comment (GCHQDeveloper404) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedAddGraphHandlerTest.java]: This graph probably doesn't need to be a FedStore this is an error carried over from before this ticket.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java [hunk 2]
```diff
@@ -97,15 +100,31 @@
...
+    private ICacheService cacheService;
     private Set<StoreTrait> traits = new HashSet<>();
     private Set<String> customPropertiesAuths;
 
-    private static List<String> getCleanStrings(final String value) {
-        final List<String> values = Arrays.asList(StringUtils.stripAll(value.split(SCHEMA_DEL_REGEX)));
-        values.remove("");
-        values.remove(null);
-        return values;
+    /**
```
Comment (GCHQDeveloper404) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java]: The re-ordering of these methods, although probably correct makes reviewing this class extremely difficult.

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java [hunk 3]
```diff
@@ -65,39 +67,115 @@
...
+        try {
+            store.initialise(FEDERATED_STORE_ID, null, storeProperties);
+            fail("Exception expected");
+        } catch (final StoreException e) {
+            assertTrue(e.getMessage().contains("No cache has been set, please check the property"));
+        }
+    }
+
+    @Test
+    public void shouldAddGraphsToCache() throws StoreException {
```
Comment (GCHQDeveloper404) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java]: This test is not checking the cache, it does not examine that cache exists, has been modified, or the cache contains the "graphToAdd". Please revise.

Comment (GCHQDeveloper404) [reply store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java]: I would want to see a test that replicates the issue explained within the ticket.

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java [hunk 4]
```diff
@@ -65,39 +67,115 @@
...
+        store.addGraphs(graphToAdd);
+
+        assertEquals(store.getGraphs(ACC_ID_1).size(), 1);
+
+        Collection<Graph> storeGraphs = store.getGraphs(null);
+        assertTrue(storeGraphs.contains(graphToAdd));
+    }
+
+    @Test
+    public void shouldAddMultipleGraphsToCache() throws StoreException {
```
Comment (GCHQDeveloper404) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java]: similar suggestion to shouldAddGraphsToCache()

**Comment by GCHQDeveloper404** [comment]
Please re-open when ready.

**Comment by p013570** [comment]
Closing while more tests are written

**Pull request merged**